apps/framework-cli/src/infrastructure/olap/clickhouse/diagnostics/stopped_operations.rs

```diff
@@ -0,0 +1,265 @@
+                    details.insert("is_readonly".to_string(), json!(true));
+                    details.insert("queue_size".to_string(), json!(queue_size));
+
+                    issues.push(Issue {
+                        severity: Severity::Error,
+                        source: "system.replicas".to_string(),
+                        component: component.clone(),
+                        error_type: "replication_stopped".to_string(),
+                        message: format!(
+                            "Replica is in read-only mode with {} items in queue. Replication may be stopped.",
+                            queue_size
+                        ),
+                        details,
+                        suggested_action: format!(
+                            "Investigate why replica is read-only. Try restarting replication: 'SYSTEM START REPLICATION QUEUES {}.{}'",
+                            db_name, component.name
+                        ),
+                        related_queries: vec![
+                            format!(
+                                "SELECT * FROM system.replicas WHERE database = '{}' AND table = '{}'",
+                                db_name, component.name
+                            ),
+                            format!(
+                                "SYSTEM START REPLICATION QUEUES {}.{}",
+                                db_name, component.name
+                            ),
+                        ],
+                    });
+                }
+            }
+        }
+
+        Ok(issues)
+    }
+}
+
+#[async_trait::async_trait]
+impl DiagnosticProvider for StoppedOperationsDiagnostic {
+    fn name(&self) -> &str {
+        "stopped_operations"
+    }
+
+    fn applicable_to(&self, _component: &Component, _engine: Option<&ClickhouseEngine>) -> bool {
+        // Applicable to all tables - we check both merges and replication
+        true
+    }
+
+    async fn diagnose(
+        &self,
+        component: &Component,
+        engine: Option<&ClickhouseEngine>,
+        config: &ClickHouseConfig,
+        _since: Option<&str>,
+    ) -> Result<Vec<Issue>, DiagnosticError> {
+        let client = ClickHouseClient::new(config)
+            .map_err(|e| DiagnosticError::ConnectionFailed(format!("{}", e)))?;
+
+        let mut issues = Vec::new();
+
+        // Check if merges are stopped for this table
+        // We can detect this by checking if there are no running merges but many parts
+        let parts_count_query = format!(
+            "SELECT count() as part_count
+             FROM system.parts
+             WHERE database = '{}' AND table = '{}' AND active = 1
+             FORMAT JSON",
+            config.db_name, component.name
+        );
+
+        debug!("Executing parts count query: {}", parts_count_query);
+
+        let parts_result = tokio::time::timeout(
+            std::time::Duration::from_secs(DIAGNOSTIC_QUERY_TIMEOUT_SECS),
```

Comment:
I feel like there is a lot of duplicated timeout code, we should consider finding a better abstraction here to improve code readability etc.